packages/ariakit-core/src/form/form-store.ts

```diff
@@ -57,9 +53,9 @@ export function get<T>(
     return defaultValue as T;
   }
   if (!rest.length) {
-    return values[key] ?? defaultValue;
+    return (values[key] ?? defaultValue) as T;
```

Comment:
Not sure if using the type assertions here is a good idea, but it doesn't affect functionality and there's no way to verify that the returned value is actually T.